pkg/controller/deliverable/reconciler.go

```diff
@@ -41,21 +42,24 @@ type Reconciler struct {
 	Realizer                realizer.Realizer
 	DynamicTracker          tracker.DynamicTracker
 	conditionManager        conditions.ConditionManager
-	logger                  logr.Logger
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.logger = logr.FromContext(ctx)
-	r.logger.Info("started")
-	defer r.logger.Info("finished")
+	log := logr.FromContext(ctx)
+	log.Info("started")
+	defer log.Info("finished")
+
+	log = log.WithValues("deliverable", req.NamespacedName)
+	ctx = logr.NewContext(ctx, log)
 
 	deliverable, err := r.Repo.GetDeliverable(ctx, req.Name, req.Namespace)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("get deliverable: %w", err)
+		log.Error(err, "failed to get deliverable")
+		return ctrl.Result{}, fmt.Errorf("failed to get deliverable [%s]: %w", err, req.NamespacedName)
```

Comment:
I don't think we usually report like this (`[..]`) - is that something we want to adopt? if so, it might be worth communicating the pattern somewhere? perhaps as part of making logging standardized, we could have an entry in CONTRIBUTING where we write down decisions like this. wdyt?

e.g., before this PR we had no matches for this type of error format:

```console
$ ag --stats "\[.*\%s\]" ./pkg
0 matches
0 files contained matches
```

as we used to mostly do like these matches

```
pkg/registrar/map_functions.go
72:             mapper.Logger.Error(err, fmt.Sprintf("could not get GVK for template: %s", templateName))
203:            return fmt.Errorf("missing apiVersion or kind: %s err: %w", obj.GetName(), err)
207:            return fmt.Errorf("unversioned object: %s", obj.GetName())
211:            return fmt.Errorf("unexpected GVK count: %s", obj.GetName())
253:            mapper.Logger.Error(err, fmt.Sprintf("could not get GVK for template: %s", templateName))
```

but now, we have a couple in this new one

```console
$ ag --stats "\[.*\%s\]" ./pkg
22 matches
7 files contained matches
133 files searched
720289 bytes searched
0.006962 seconds
```

Reply:
Yes this needs intentional discussion. Our idea is to put any key or simple key-value information in `[]` and list/complex data after a `:`

Some people use backticks, but whatever we use it needs to be obvious we are refering to value types.

personally I like Emily's Square Brackets (as they will forever be known, ESB for short) and naturally we want to update contributing with this information (and many other bits) after we ratify it